presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -677,7 +680,19 @@ private TaskContext createTaskContext()
     private static LookupSourceFactory buildHash(boolean parallelBuild, TaskContext taskContext, List<Integer> hashChannels, RowPagesBuilder buildPages, Optional<InternalJoinFilterFunction> filterFunction)
     {
         Optional<JoinFilterFunctionFactory> filterFunctionFactory = filterFunction
-                .map(function -> ((session, addresses, channels) -> new StandardJoinFilterFunction(function, addresses, channels)));
+                .map(function -> new JoinFilterFunctionFactory() {
+                    @Override
+                    public JoinFilterFunction create(ConnectorSession session, LongArrayList addresses, List<List<Block>> channels)
+                    {
+                        return new StandardJoinFilterFunction(function, addresses, channels, Optional.empty());
+                    }
+
+                    @Override
+                    public Optional<SortChannelExtractor.SortChannel> getSortChannel()
+                    {
+                        return Optional.empty();
```

Comment:
I think is not worth testing, since writing such test would be quite painful. 

There are unit tests for both `SortedPositionLinks` and `SortChannelExtractor` and integration between `JoinHash` and `PositionLinks` interface is tested here regardless of this `Optional.empty()`. Also there are integration tests in `AbstractTestQueries` for this .